iReading styles integration

nathanstitt opened the pull request:
![screen shot 2015-05-17 at 10 39 26 pm](https://cloud.githubusercontent.com/assets/79566/7674107/0d2cb2a4-fce6-11e4-9db5-e62c407d3b2e.png)

![screen shot 2015-05-17 at 10 40 08 pm](https://cloud.githubusercontent.com/assets/79566/7674108/11119722-fce6-11e4-9917-799280e81ce4.png)

![screen shot 2015-05-18 at 11 42 29 am](https://cloud.githubusercontent.com/assets/79566/7685865/16112690-fd57-11e4-86d4-4b2c19eb9a40.png)

pandafulmanda commented:
:+1:

philschatz commented on resources/styles/components/task-step/reading/book-content/index.less:
```diff
@@ -25,20 +28,43 @@
+
+    > [data-type=exercise] {
+
+      content: "";
```
What does this line do?

philschatz commented on resources/styles/components/task-step/reading/book-content/index.less:
```diff
@@ -25,20 +28,43 @@
+      }
+      
+    }
+    .problem, .solution, .commentary {
```
Can you include each selector on a separate line?

nathanstitt replied:
That entire block is gone now.  The new mockups aren't using 0 margin on the left/right sides so there's now no need to have the margins set to 0 and then add it back on all the elements.

philschatz commented on resources/styles/components/task-step/reading/book-content/index.less:
```diff
@@ -25,20 +28,43 @@
 }
 
+.note > .title, .commentary [data-type=title] {
+  color: #77af41;
```
Should this color be defined at the top of the file?

philschatz commented on resources/styles/components/task-step/reading/book-content/tips-for-success.less:
```diff
@@ -1,19 +1,38 @@
-    margin: 0 500px 15px 0;
+    height: 35px;
+    content: attr(data-label);
+    background: @tutor-quaternary url("icon-light-bulb.svg");
```
Does splitting this into `background-color: ` and `.tutor-reading-step-banner-icon(...)` not work?

nathanstitt replied:
Yeah, I started with the `tutor-reading-step-banner-icon` mixin but ended up changing every single thing on it for the tips-for-success, so I just made it be it's own thing.  It's a different size, color, and needs to have different text flow semantics.

nathanstitt commented:
@Fredasaurus / @philschatz  I'd like your opinion of the video steps that have text before the video and no titles like this one:

![screen shot 2015-05-18 at 12 01 10 pm](https://cloud.githubusercontent.com/assets/79566/7685887/42c3dcbe-fd57-11e4-9bf9-8752e9f9c3a1.png)

Because of the way the HTML is laid out, there's not an element that I can attach the frame to in order to style them like the other video steps such as this:

![screen shot 2015-05-18 at 11 43 01 am](https://cloud.githubusercontent.com/assets/79566/7685903/5adb8b12-fd57-11e4-9c48-7efa9c7e8e2f.png)

Is it important that the video have the frame at all times?  If it is, I'll have to add some code to manipulate the DOM manually after rendering to make sure that all the videos have a containing element.

nathanstitt commented:
Renamed title since this PR has grown to include styles other than worked examples, it now contains code to implement the latest mockup since when I changed one thing it'd ripple out and affect other step types as well.

nathanstitt commented:
The difference between the two video steps is that one has a wrapping div[data-type=media] container. 

The presence of that div allows us to position the wrapper as `position: relative`, and the iframe as `position: absolute` with it's right/left sides set at 3px so the drop shadow indent effect will appear. [full styles](https://github.com/openstax/tutor-js/blob/worked-example-polish/resources/styles/mixins.less#L26-L40)


The wrapped version:

```html
    <div data-type="note" data-label="Watch Physics">
      <div data-type="title" class="title">Newton’s Third Law of Motion</div>
      <div data-type="media">
        <iframe width="660" height="371.4" src="https://www.khanacademy.org/embed_video?v=By-ggTfeuJU"></iframe>
      </div>
    </div>
```

And the unwrapped:
```html
    <div data-type="note" data-label="Watch Physics">
      <div data-type="title" class="title">A Note called Watch Physics with YouTube Embedded</div>
      <p>Lorem ipsum dolo.</p>
      <iframe src="https://www.youtube.com/embed/40ETbLVkLKc"></iframe>
    </div>
```

Fredasaurus commented:
"drop shadow indent effect" What is this?

nathanstitt commented:
It's the subtle little frame that videos in the mockups have that makes them appear indented.  Maybe you didn't intend for that effect?  If not, we can just make the iframe 100% width and be done.

![screen shot 2015-05-18 at 2 11 37 pm](https://cloud.githubusercontent.com/assets/79566/7688138/d77a4c84-fd67-11e4-874b-3e79b8f077cf.png)

Fredasaurus commented:
Ah, good eye. Not even sure where that came from. We can remove it, thanks.

nathanstitt commented:
Ok, disabled the drop shadow by default.  I left as a mixin in case we need to customize further later
